### PR TITLE
Infrastructure for using active enumerators in sygus modules

### PR DESCRIPTION
**Pull request by ajreynol**
This introduces a notion of "actively-generated" vs "passively-generated" enumerators, in preparation for variable agnostic enumerators. It updates the logic of sygus modules to account for this kind of enumeration.

### CODE REVIEW

**Comment by HanielB** [comment]
clang-format

File: src/theory/quantifiers/sygus/cegis.cpp [hunk 1]
```diff
@@ -100,15 +101,45 @@ void Cegis::getTermList(const std::vector<Node>& candidates,
...
+  // apply this form of refinement if any (relevant) enumerator in candidates is
+  // "actively generated" (see TermDbSygs::isPassiveEnumerator), since its
+  // model values are themselves interpreted as classes of solutions.
+  bool doGen = true;
+  for( const Node& v : candidates )
+  {
+    // if it is relevant to refinement
+    if( d_refinement_lemma_vars.find(v)!=d_refinement_lemma_vars.end() )
+    {
+      if( !d_tds->isPassiveEnumerator(v) )
```
[the remark this answers is not in the document]
Comment (ajreynol) [reply src/theory/quantifiers/sygus/cegis.cpp]: Why? Checking d_refinement_lemma_vars is a single map lookup, whereas isPassiveEnumerator might be more complicated.

[the remark this answers is not in the document]
Comment (HanielB) [reply src/theory/quantifiers/sygus/cegis.cpp]: you're right, I got confused

[the remark this answers is not in the document]
Comment (HanielB) [reply src/theory/quantifiers/sygus/cegis.cpp]: But do join the `if`s, please. They are fully independent

File: src/theory/quantifiers/sygus/term_database_sygus.h [hunk 2]
```diff
@@ -102,7 +102,8 @@ class TermDbSygus {
    * are obtained by looking at its model value only. For passively-generated
    * enumerators, it is the responsibility of the user of that enumerator (say
    * a SygusModule) to block the current model value of it before asking for
-   * another value.
+   * another value. By default, the Cegis module uses passively-generated
```
Comment (HanielB) [inline src/theory/quantifiers/sygus/term_database_sygus.h]: add a "for example" and the reason why

File: src/theory/quantifiers/sygus/cegis.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (HanielB) [reply src/theory/quantifiers/sygus/cegis.cpp]: actually, yeah, leave as it is

**Pull request merged**